hazelcast/src/test/java/com/hazelcast/cluster/TcpIpJoinTest.java

```diff
@@ -224,4 +232,103 @@ public void test_whenIncompatibleJoiners() throws Exception {
 
         assertIncompatible(config1, config2);
     }
+
+    @Test
+    public void test_tcpIpJoinerRemembersJoinedMemberAddresses() {
+        Config config1 = smallInstanceConfig()
+                .setProperty("hazelcast.merge.first.run.delay.seconds", "10")
+                .setProperty("hazelcast.merge.next.run.delay.seconds", "10");
+        JoinConfig join1 = config1.getNetworkConfig().getJoin();
+        join1.getMulticastConfig().setEnabled(false);
+        TcpIpConfig tcpIpConfig1 = join1.getTcpIpConfig();
+        tcpIpConfig1.setEnabled(true);
+
+        Config config2 = smallInstanceConfig();
+        JoinConfig join2 = config2.getNetworkConfig().setPort(8899).getJoin();
+        join2.getMulticastConfig().setEnabled(false);
+        TcpIpConfig tcpIpConfig2 = join2.getTcpIpConfig();
+        tcpIpConfig2.setEnabled(true).addMember("127.0.0.1:5701");
+
+        Config config3 = smallInstanceConfig();
```

Comment:
This may be extracted to a method that takes only port as an argument. Then instead of 24 lines you will have 3 here :)

Reply:
Done 👍